python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -577,13 +577,23 @@ def __init__(
         self.options = options
         self.name = name
         self.children = children
+        self._validate_input()
+
+    def _validate_input(self):
         if self.name not in (
             pl_expr.StringFunction.Lowercase,
             pl_expr.StringFunction.Uppercase,
             pl_expr.StringFunction.EndsWith,
             pl_expr.StringFunction.StartsWith,
+            pl_expr.StringFunction.Contains,
         ):
             raise NotImplementedError(f"String function {self.name}")
+        if self.name == pl_expr.StringFunction.Contains:
+            _, strict = self.options
+            if not strict:
+                raise NotImplementedError("strict=False not supported in contains")
```

Comment:
I think we're now too strict (hoho), in two ways:

1. `strict=False` is probably meaningless if we have a literal pattern
2. if we have a literal pattern, then we can have an arbitrary column as the pattern object.

Reply:
So how about:

```
literal, strict = self.options
if not literal:
    if not strict:
        raise NotImplementedError("f{strict=} is not supported for regex contains")
    if not isinstance(self.children[1], Literal):
        raise NotImplementedError("Regex contains only supports a scalar pattern")
```
?